src/Abp/Runtime/Caching/AbpCacheBase.cs

```diff
@@ -32,104 +33,87 @@ protected AbpCacheBase(string name) : base(name)
 
         public virtual TValue Get(TKey key, Func<TKey, TValue> factory)
         {
-            TValue item = default(TValue);
 
-            try
+            if (TryGetValue(key, out TValue value))
             {
-                item = GetOrDefault(key);
-            }
-            catch (Exception ex)
-            {
-                Logger.Error(ex.ToString(), ex);
+                return value;
             }
 
-            if (item == null)
+            lock (SyncObj)
             {
-                lock (SyncObj)
+                if (TryGetValue(key, out value))
+                {
+                    return value;
+                }
+
+                var generatedValue = factory(key);
+                if (!EqualityComparer<TValue>.Default.Equals(generatedValue, default))
```

Comment:
Let's define a `protected virtual bool ShouldSet(TValue generatedValue)` that can be overridden to cache generated values that are (or happen to be) equal to `default`.

Reply:
i would go for `bool IsDefaultValue(TValue value)` instead

Reply:
`IsDefaultValue(value)` doesn't convey the same meaning as `ShouldSet(generatedValue)` (or `ShouldCache(generatedValue)`).

Reply:
@ismcagdas we might want to get this change in for 6.0 as well. I will create a separate PR for this soon.

Reply:
@ryancyq it should be already done. You can create a new issue if you think something is missing 👍